### ISSUE TITLE
init repo

### ISSUE THREAD

**Issue by beck** [yola/eslint-config-yola#1]
Port the pull from:
https://github.com/yola/jsgreat/pull/34/files

eslint does not support sharable config that uses plugins:
https://github.com/eslint/eslint/issues/3458

I think the right approach for us is:
* extend a version of airbnb without plugins, `{ "extends": "airbnb/base" }`
* restore the plugins we want using this approach: https://github.com/mmazzarolo/eslint-plugin-react-app/blob/2b0da6a2dcf53803a66d10fc22cfaf043786b609/index.js

**Comment by RusinovAnton** [comment]
@beck seems like we can't setup extending airbnb config without using plugins because `eslint-config-airbnb-base` depends at least on `eslint-plugin-import`.

https://github.com/airbnb/javascript/tree/master/packages/eslint-config-airbnb-base#eslint-config-airbnb-base-1

**Comment by RusinovAnton** [comment]
Also I'd like to test it from npm repo rather than locally since command `npm link eslint-config-yola` doesn't resolve its dependencies. eg
```
npm run eslint
> sitebuilderui@0.0.0 eslint /home/rusinov/yola/sitebuilderui
> eslint src/apps/**/*.js

Cannot find module 'eslint-config-airbnb-base'
Referenced from: /home/rusinov/yola/eslint-config-yola/index.js
Referenced from: /home/rusinov/yola/sitebuilderui/.eslintrc
Error: Cannot find module 'eslint-config-airbnb-base'
Referenced from: /home/rusinov/yola/eslint-config-yola/index.js
Referenced from: /home/rusinov/yola/sitebuilderui/.eslintrc
```

So I want to publish dev version of package. I think there should be "Upload library" jenkins task as for other libs but how do I setup that?

**Comment by RusinovAnton** [comment]
> restore the plugins we want using this approach: https://github.com/mmazzarolo/eslint-plugin-react-app/blob/2b0da6a2dcf53803a66d10fc22cfaf043786b609/index.js

So we need to have `eslint-config-yola` to define rules and then `eslint-plugin-yola` that will depend on our `eslint-config-yola` and wrap rules from needed plugins.

**Comment by RusinovAnton** [comment]
On friday I've published `jsgreat@0.2.0-dev.1`  https://github.com/yola/jsgreat/pull/34/commits/bd0360db114304563f3b02852ed7c7ecff03f04f with eslint plugins declared in `dependencies` rather than in `devDependencies` so plugins then were installed with `jsgreat` package. I've tried to use it in sitebuilderui and were able to use shareable eslint config from `jsgreat` without declaring that plugins in sitebuilderui itself.  
So I mean maybe we don't need extra package `eslint-plugin-yola` to resolve plugins deps?

**Comment by beck** [comment]
Publishing should not resolve the dependencies, if it does then there is something else wrong (either with the source or your local dev setup).

```
"extends": "plugin:yola"
```

Instead of using a name string for the `extends` value, will a path or an imported module work? That would save us from having to maintain two repos.

**Comment by RusinovAnton** [comment]
> Publishing should not resolve the dependencies   

Why? Isn't npm should install dependencies of dependencies in package.json.


To sum up everything I wrote above, I think we should have single repo for shareable config `eslint-config-yola` and declare eslint-plugins in its `package.json` as `dependencies`.
The config itself may look like this:
```
module.exports = {
  "extends": "airbnb-base",
  "plugins": [
    "import"
  ]
};
```

Then we add it as `devDependency` to our `sitebuilderui` with command `npm i -D eslint-config-yola` which should resolve eslint-plugins dependencies too. Then in `sitebuilderui/.eslintrc` we add
```
{ "extends": "yola" }
```
which is alias for 
```
{ "extends": "eslint-config-yola" }
```
and run `eslint -c .eslintrc [...files]` for the win.

My problem here is that when I ran `npm link` instead of `npm install` it didn't install plugins that were declared as dependencies in `eslint-config-yola/package.json`.
